### ISSUE TITLE
如果训练的验证码尺寸不一样，应该如何修改 CNN 模型呢？

### ISSUE THREAD

**Issue by GeekNOmore** [ice-tong/pytorch-captcha#5]
比如如果获取的是 150*60 的验证码，如何修改 CNN 模型呢？

**Comment by GeekNOmore** [comment]
`import torch.nn as nn

class CNN(nn.Module):
    def __init__(self, num_class=36, num_char=4):
        super(CNN, self).__init__()
        self.num_class = num_class
        self.num_char = num_char
        self.conv = nn.Sequential(
                #batch*3*150*60
                nn.Conv2d(3, 16, 3, padding=(1, 1)),
                nn.MaxPool2d(2, 2),
                nn.BatchNorm2d(16),
                nn.ReLU(),
                #batch*16*75*30
                nn.Conv2d(16, 64, 3, padding=(1, 1)),
                nn.MaxPool2d(2, 2),
                nn.BatchNorm2d(64),
                nn.ReLU(),
                #batch*64*45*15
                nn.Conv2d(64, 512, 3, padding=(1, 1)),
                nn.MaxPool2d(2, 2),
                nn.BatchNorm2d(512),
                nn.ReLU(),
                #batch*512*22*7
                nn.Conv2d(512, 512, 3, padding=(1, 1)),
                nn.MaxPool2d(2, 2),
                nn.BatchNorm2d(512),
                nn.ReLU(),
                #batch*512*11*3
                )
        self.fc = nn.Linear(512*11*3, self.num_class*self.num_char)
        
    def forward(self, x):
        x = self.conv(x)
        x = x.view(-1, 512*11*3)
        x = self.fc(x)
        return x`
这样修改之后，得到 runtime error
shape '[-1, 16896]' is invalid for input of size 1769472
如果直接修改 512*11*3 成 1769472，会爆显存...

**Comment by GeekNOmore** [comment]
修改了 batch size，不爆了。但是我依然没懂 shape[] 里的数字是怎么计算的...

**Comment by ice-tong** [comment]
你可以在x = self.conv(x)后面打印一下x的shape，print(x.shape)，就可以知道全连接的维度应该修改成多大了。


GeekNOmore ***@***.***>于2021年5月23日 周日13:10写道：
